pkg/xds/generator/egress/generator.go

```diff
@@ -123,3 +123,11 @@ func buildDestinations(
 
 	return destinations
 }
+
+func isReachableFromZone(endpoint *core_xds.Endpoint, zone string) bool {
+	return endpoint.Tags[mesh_proto.ZoneTag] == "" || endpoint.Tags[mesh_proto.ZoneTag] == zone
+}
+
+func isOnlyReachableFromOtherZones(endpoint *core_xds.Endpoint, localZone string) bool {
+	return endpoint.Tags[mesh_proto.ZoneTag] != "" && endpoint.Tags[mesh_proto.ZoneTag] != localZone
+}
```

Comment:
Instead of rebuilding ifs here, would it be possible to provide this information in Endpoint#ExternalService when we create endpoints?